blockchain/message_pool/src/msgpool.rs

```diff
@@ -373,10 +374,7 @@ where
             return Ok(());
         }
 
-        let umsg = msg.message().marshal_cbor()?;
-        msg.signature()
-            .verify(umsg.as_slice(), msg.from())
-            .map_err(Error::Other)?;
+        msg.verify().map_err(Error::Other)?;
```

Comment:
Should this error type be `InvalidFromAddr` rather than `Other`? It seems the `verify` fn would fail if the `from` address is not the correct address which seems more of an `InvalidFromAddr` error type.

Reply:
That's not the only error that can happen with signature validation. You also lose the error message for no benefit. 

Side note, I don't like the error type in this crate, would be nice to improve later.